jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import java.awt.event.ActionEvent
+import javax.swing.Action
+import javax.swing.JComponent
+import software.aws.toolkits.resources.message
+import java.time.temporal.ChronoUnit
+import java.util.Calendar
+import java.util.Date
+
+class QueryEditorDialog(
+    private val project: Project,
+    private val lGroupName: String,
+    private val absoluteTimeSelected: Boolean = false
+
+) : DialogWrapper(project) {
+    constructor(project: Project, logGroupName: String) :
+        this(project = project, lGroupName = logGroupName)
+
+    private val view = QueryEditor(project)
+    private val qQueryingLogGroupApiCall = QueryingLogGroups(project)
+    private val action: OkAction = QueryLogGroupOkAction()
+    private val validator = QueryEditorValidator
+    init {
+        super.init()
+        title = "Query Log Groups"
+        view.absoluteTimeRadioButton.addActionListener {
+            view.startDate.isEnabled = true
+            view.endDate.isEnabled = true
+            view.relativeTimeNumber.isEnabled = false
+            view.relativeTimeUnit.setEnabled(false)
+        }
+        view.relativeTimeRadioButton.addActionListener {
+            view.startDate.isEnabled = false
+            view.endDate.isEnabled = false
+            view.relativeTimeNumber.isEnabled = true
+            view.relativeTimeUnit.setEnabled(true)
+        }
+        view.queryLogGroupsRadioButton.addActionListener {
+            view.queryBox.isEnabled = true
+            view.querySearchTerm.isEnabled = false
+        }
+        view.searchTerm.addActionListener {
+            view.queryBox.isEnabled = false
+            view.querySearchTerm.isEnabled = true
+        }
+    }
+    override fun createCenterPanel(): JComponent? = view.queryEditorBasePanel
+    override fun doValidate(): ValidationInfo? = validator.validateEditorEntries(view)
+    override fun getOKAction(): Action = action
+    override fun doCancelAction() {
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        // Do nothing, close logic is handled separately
+    }
+
+    private fun getRelativeTime(unitOfTime: String, relTimeNumber: Long): StartEndDate {
+        val endDate = Calendar.getInstance().toInstant()
+        val startDate = when (unitOfTime) {
+            "Minutes" -> endDate.minus(relTimeNumber, ChronoUnit.MINUTES)
```

Comment:
When there are strings like this, they should be put in a central place so they can be referenced by a const in both places

Reply:
Assigned the strings to variables.